### PR TITLE
rpm: provide systemd and microshift defaults

### PR DESCRIPTION
**Pull request by frzifus**
Implements the in [[link](https://docs.google.com/document/d/1mOOETdfluOpL2GT4CYxVYo3O5Tw-BURUpC6EZD3YTXI)] proposed change.

For example, to foward journald logs to AWS cloudwatch, a user only needs to create file `/etc/opentelemetry-collector/configs/10-cloudwatch-export.yaml` with the following content:
```yaml
exporters:
  awscloudwatchlogs:
    log_group_name: "testing-logs-emf"
    log_stream_name: "testing-integrations-stream-emf"

service:
  pipelines:
    logs:
      receivers: [journald]
      exporters: [awscloudwatchlogs]
```
Afterwards, the opentelemetry-collector service must be restarted via systemd.

```bash
systemctl restart opentelemetry-collector.service
```

---

The Package will also install the manifests for an OpenTelemetry Collector on microshift. Its used to accept telemetry data from inside of microshift. (e.g. scraping the kubelet api, collecting metrics, logs and events.)

The manifests will be installed in `/etc/microshift/manifests/opentelemetry-collector`. If microshift is not present on the machine, this will have no effect.

An example configuration would look similar to the cloudwatch config. But instead of accepting logs from the `journald`  receiver, we would accept data from the `otlp/microshift` receiver.

```yaml
service:
  pipelines:
    logs:
      receivers: [otlp/microshift]
      exporters: [debug]
    metrics:
      receivers: [otlp/microshift]
      exporters: [debug]
    traces:
      receivers: [otlp/microshift]
      exporters: [debug]
```
**NOTE**: Debug can be replaced by any kind of exporter that can registered to a pipeline type.

cc @miyunari @fkolwa @major @mparram 

---

Build:
https://copr.fedorainfracloud.org/coprs/miyunari/redhat-opentelemetry-collector/build/7390581/

### CODE REVIEW

**Comment by frzifus** [comment]
This PR needs to be rebased once https://github.com/os-observability/redhat-opentelemetry-collector/pull/66 got merged.

**Comment by frzifus** [comment]
cc @examon

**Comment by frzifus** [comment]
The package seems now to work as expected. I used to copr build for a short test.

For testing I created a `/etc/opentelemetry-collector/configs/10-debug-export.yaml` file with the following content:
```yaml
---
service:
  pipelines:
    logs:
      receivers: [journald]
      exporters: [debug]
    metrics:
      receivers: [hostmetrics, hostmetrics/disk]
      exporters: [debug]
```

After restarting the collector  instance, I was finally able to see the data the journald logs.

File: collector.spec.in [hunk 1]
```diff
@@ -33,22 +37,68 @@ mkdir -p _build/bin
 %setup -q -n %{name}-%{version}
 
 %build
+
 go build -v -buildmode pie -mod vendor -o %{gobuilddir}/bin/opentelemetry-collector
+mv opentelemetry-collector-with-options %{gobuilddir}/bin/opentelemetry-collector-with-options
```
Comment (FKolwa) [inline collector.spec.in:42]: I ran the build through fedpkg and I had no issues replacing this line and adding it to line 61.
```suggestion
install -m 0755 -p opentelemetry-collector-with-options %{buildroot}%{_bindir}/
```
I remember that you had issues using install to copy this file to bin but there might have been a different root cause for this.

**Review by FKolwa — CHANGES_REQUESTED** [review]
I would try and fix the install of "opentelemetry-collector-with-options", other than that: Great job! 

-----------------------------------------------
There are some rpmlint warnings (can be ignored) and errors which should be handled in the longrun:
```
opentelemetry-collector.x86_64: W: unstripped-binary-or-object /usr/bin/opentelemetry-collector
opentelemetry-collector.x86_64: W: only-non-binary-in-usr-lib
opentelemetry-collector.x86_64: W: non-conffile-in-etc /etc/microshift/manifests/opentelemetry-collector/kustomization.yaml
opentelemetry-collector.x86_64: W: non-conffile-in-etc /etc/microshift/manifests/opentelemetry-collector/manifest.yaml
opentelemetry-collector.x86_64: W: non-conffile-in-etc /etc/opentelemetry-collector/configs/00-default-receivers.yaml
opentelemetry-collector.src: W: no-url-tag
opentelemetry-collector.x86_64: W: no-url-tag
opentelemetry-collector.x86_64: W: no-manual-page-for-binary opentelemetry-collector
opentelemetry-collector.x86_64: W: no-manual-page-for-binary opentelemetry-collector-with-options
collector.spec: W: invalid-url Source0: opentelemetry-collector-0.99.0.tar.gz
collector.spec: W: invalid-url Source0: opentelemetry-collector-0.99.0.tar.gz
opentelemetry-collector.src: E: invalid-spec-name
opentelemetry-collector.src: E: description-line-too-long Collector with the supported components for a Red Hat build of OpenTelemetry product
opentelemetry-collector.x86_64: E: description-line-too-long Collector with the supported components for a Red Hat build of OpenTelemetry product
opentelemetry-collector.x86_64: W: dangerous-command-in-%postun userdel
```

File: collector.spec.in (hunk 1 above)
Comment (frzifus) [reply collector.spec.in:42]: But the final installation happens then in the install section, right?
image
Means we would call install twice. Isnt that a bit misleading?

Comment (FKolwa) [reply collector.spec.in:42]: yes that is correct so basically this:
```
# install files
install -p -m 0644  ./00-default-receivers.yaml %{buildroot}%{_sysconfdir}/opentelemetry-collector/configs/00-default-receivers.yaml
install -p -m 0644  ./opentelemetry-collector.service %{buildroot}%{_unitdir}/%{name}.service
install -p -m 0644  ./microshift-opentelemetry-gateway.yaml %{buildroot}%{_sysconfdir}/microshift/manifests/opentelemetry-collector/manifest.yaml
install -p -m 0644  ./kustomization.yaml %{buildroot}%{_sysconfdir}/microshift/manifests/opentelemetry-collector/kustomization.yaml

install -m 0755 -vd                     %{buildroot}%{_bindir}
install -m 0755 -vp %{gobuilddir}/bin/* %{buildroot}%{_bindir}/
install -m 0755 -p opentelemetry-collector-with-options %{buildroot}%{_bindir}/
```
The line above shouldn't be put into line 42.

**Review by FKolwa — APPROVED** [review]
Looks good to me. Once again: Great work!

**Pull request merged**